Editor/Extensions/SerializedPropertyExtensions.cs

```diff
@@ -41,6 +41,14 @@ public static T GetValue<T>(this SerializedProperty property) where T : class
             return (T)obj;
         }
 
+        private static void ParseFieldStructure(string input, out string fieldName, out int index)
+        {
+            int startIndex = input.IndexOf('[');
+            int endIndex = input.IndexOf(']');
```

Comment:
```suggestion
            int endIndexSearchStart = Mathf.Min(startIndex + 1, input.Length - 1);
            int endIndexSearchCount = Mathf.Min(9, input.Length - startIndex - 1);
            int endIndex = input.IndexOf(']', endIndexSearchStart, endIndexSearchCount );
```
The end index is supposed to always be shortly after the start index, right?

So it would be good to start looking right after the start index. You could also limit it to a few characters (say, 8).

Then we have to make sure it doesn't exceed the length of the string because that would [throw an ArgumentOutOfRangeException apparently](https://learn.microsoft.com/en-us/dotnet/api/system.string.indexof?view=net-8.0)

This would die a horrible death if `startIndex` is -1 but I think it's supposed to always be there, right?

Maybe using a search count is unnecessary hassle. At the very least you should be able to use a start index though 😄